src/lib/components/Modal/Modal.jsx

```diff
@@ -1,145 +1,152 @@
+  position,
+  size,
+) => {
+  const sizeClass = (modalSize) => {
+    if (modalSize === 'small') {
+      return styles.isRootSizeSmall;
+    }
 
-    this.childrenWrapperRef = React.createRef();
+    if (modalSize === 'medium') {
+      return styles.isRootSizeMedium;
+    }
 
-    this.keyPressHandler = this.keyPressHandler.bind(this);
-  }
+    if (modalSize === 'large') {
+      return styles.isRootSizeLarge;
+    }
+
+    if (modalSize === 'fullscreen') {
+      return styles.isRootSizeFullscreen;
+    }
+
+    return styles.isRootSizeAuto;
+  };
+
+  const positionClass = (modalPosition) => {
+    if (modalPosition === 'top') {
+      return styles.isRootPositionTop;
+    }
 
-  componentDidMount() {
-    const {
-      autoFocus,
-      primaryButtonRef,
-    } = this.props;
+    return styles.isRootPositionCenter;
+  };
 
-    window.document.addEventListener('keydown', this.keyPressHandler, false);
+  return (
+    <div
+      className={styles.backdrop}
+      id={id}
+      onClick={() => {
+        if (closeButtonRef?.current != null) {
+          closeButtonRef.current.click();
+        }
+      }}
+      role="presentation"
+    >
+      <div
+        className={classNames(
+          styles.root,
+          sizeClass(size),
+          positionClass(position),
+        )}
+        onClick={(e) => {
+          e.stopPropagation();
+        }}
+        role="presentation"
+        ref={childrenWrapperRef}
+      >
+        {children}
+      </div>
+    </div>
+  );
+};
+
+export const Modal = ({
+  autoFocus,
+  children,
+  closeButtonRef,
+  id,
+  portalId,
+  position,
+  primaryButtonRef,
+  size,
+}) => {
+  const childrenWrapperRef = useRef();
+
+  const keyPressHandler = (e) => {
+    if (e.keyCode === 27 && closeButtonRef?.current != null) {
+      closeButtonRef.current.click();
+    }
+
+    if (e.keyCode === 13 && e.target.nodeName !== 'BUTTON' && primaryButtonRef?.current != null) {
+      primaryButtonRef.current.click();
+    }
+  };
+
+  useEffect(() => {
+    window.document.addEventListener('keydown', keyPressHandler, false);
 
     // If `autoFocus` is set to `true`, following code finds first form field element
     // (input, textarea or select) or primary button and auto focuses it. This is necessary
     // to have focus on one of those elements to be able to submit form by pressing Enter key.
     if (autoFocus) {
-      if (this.childrenWrapperRef?.current != null) {
-        const childrenWrapperElement = this.childrenWrapperRef.current;
+      if (childrenWrapperRef?.current != null) {
+        const childrenWrapperElement = childrenWrapperRef.current;
         const childrenElements = childrenWrapperElement.querySelectorAll('*');
         const formFieldEl = Array.from(childrenElements).find(
           (element) => ['INPUT', 'TEXTAREA', 'SELECT'].includes(element.nodeName),
         );
 
         if (formFieldEl) {
           formFieldEl.focus();
-          return;
+          return () => {};
```

Comment:
~~This is exactly the same case as in @dacerondrej PR we dealt with yesterday. I think it would be better to return `undefined` here because:~~
1. ~~I think i better communicates the intention~~
2. ~~I would expect it to be more performant (I dont know though)~~

On second thought I think the `return` should not be here at all. As it is, in case I have form elements to autofocus, the `keyDown` event handler does not get cleared on unmount.